### ISSUE TITLE
Fuzzy Search Question

### ISSUE THREAD

**Issue by Matthew149Guy** [dj-nitehawk/MongoDB.Entities#60]
Hi, I have a question regarding Fuzzy Search results, I know we have the SortByRelevance() method available, but I was wondering if there was a way to get the actual numeric representation of the Levenshtein distance for any given match within the library - or would I need to implement this myself. The reason I ask is I'd like to filter out fuzzy matches by tuning the maximum distance.

Thanks

**Comment by dj-nitehawk** [comment]
just pushed an update with an optional parameter `maxDistance` for SortByRelevance(). you can now specify the cutoff distance. any item that is greater than the specified distance will be excluded from the resulting list.

also worth mentioning, you might be able to get some additional mileage by using the `DB.Find<T>().SortByTextScore()` together with `.Skip()`, `Limit()` to shortlist the results on the server-side before downloading the items to your client/app.

haven't tried that myself, but it might work for your dataset.

**Comment by Matthew149Guy** [comment]
Wow, that was quick. I'll give it whirl and let you know how it work out. Also, thanks for the advice on using SortByTextScore()

Thanks

**Comment by Matthew149Guy** [comment]
Hi, apologies I got so engrossed in working on my project I forgot to give feedback. The method works lovely, and actually highlighted I needed something more sophisticated. I learned what I needed to do was find the Levenshtein distance between the search metaphone and data metaphones rather than the plain string. Luckily you provide a handy string extension method (ToDoubleMetaphone()) which allows me to do this, in a slight more round-about way, but it works perfectly for me. Again, thanks.
